core/tests/ui/e2e/order.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { expect, test } from '~/tests/fixtures';
+
+function formatAmount(amount: number): string {
+  return new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+  }).format(amount);
+}
+
+function formatDate(s: string): string {
+  return new Date(s).toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  });
+}
+
+test('Orders page has empty state', async ({ page, account }) => {
+  const customer = await account.create();
+
+  await customer.login();
+
+  await page.goto('/account/orders/');
+
+  await expect(page.getByRole('heading', { name: 'Orders' })).toBeVisible();
+  await expect(page.getByText('No orders yet')).toBeVisible();
+});
+
+test('Order details are visible on Orders page', async ({ page, account, product, order }) => {
+  const customer = await account.create();
+
+  await customer.login();
+
+  const orderProduct = await product.get(88);
+
+  const orderDetails = await order.create(customer.id, 88);
+
+  await page.goto('/account/orders/');
+
+  const formattedTotal = formatAmount(parseFloat(orderDetails.total_inc_tax));
+  const formattedProductPrice = formatAmount(orderProduct.price ?? 0);
+  const formattedDate = formatDate(orderDetails.date_created);
+
+  await expect(page.getByRole('link', { name: `Order # ${orderDetails.id}` })).toBeVisible();
+  await expect(page.getByText('Order placed')).toBeVisible();
+  await expect(page.getByText(formattedDate)).toBeVisible();
+  await expect(page.getByText('Total')).toBeVisible();
+  await expect(page.getByText(formattedTotal)).toBeVisible();
+  await expect(page.getByText(orderDetails.status)).toBeVisible();
+  await expect(page.getByRole('link', { name: orderProduct.name })).toBeVisible();
+  await expect(page.getByText(formattedProductPrice)).toBeVisible();
+});
```

Comment:
Can we get rid of the `ProductFactory` and inline the product information? I see the test looking like this:
```ts
test('Order details are visible on Orders page', async ({ page, account, order }) => {
  const customer = await account.create();

  await customer.login();

  // Customer id last and make it optional – in case we want to test guest shopper orders?
  const orderDetails = await order.create(88, customer.id);

  await page.goto('/account/orders/');

  await expect(page.getByRole('link', { name: `Order # ${orderDetails.id}` })).toBeVisible();
  await expect(page.getByText('Order placed')).toBeVisible();
  await expect(page.getByText(orderDetails.date_created)).toBeVisible();
  await expect(page.getByText('Total')).toBeVisible();
  await expect(page.getByText(orderDetails.total_inc_tax)).toBeVisible();
  await expect(page.getByText(orderDetails.status)).toBeVisible();
  await expect(page.getByRole('link', { name: ''[Sample] Chemex Coffeemaker 3 Cup
 })).toBeVisible();
  await expect(page.getByText('$49.50')).toBeVisible();
});
```

Also, we haven't been testing internationalization yet, so I don't think we need to format the dates/currencies yet. I think we might want to hold off till we support Single-channel multi-locale in Catalyst for this.

Reply:
format functions needed to convert "Thu, 03 Oct 2024 09:41:52 +0000" to "October 3, 2024" and "59.4000" to "$59.40"

Reply:
Ok, I'll update it

Reply:
done ✅

Reply:
>format functions needed to convert "Thu, 03 Oct 2024 09:41:52 +0000" to "October 3, 2024" and "59.4000" to "$59.40"

Ahh okay! Makes sense!